config.go

```diff
@@ -109,38 +110,28 @@ func (c *RuntimeConfig) WithFeatureSignExtensionOps(enabled bool) *RuntimeConfig
 	return ret
 }
 
-// Module is a WebAssembly 1.0 (20191205) module to instantiate.
-type Module struct {
-	name   string
-	module *internalwasm.Module
-}
-
-// WithName configures the module name. Defaults to what was decoded from the module source.
+// Binary is a WebAssembly 1.0 (20191205) module ready to be instantiated (Runtime.InstantiateModule) as an api.Module.
 //
-// If the source was in WebAssembly 1.0 (20191205) Binary Format, this defaults to what was decoded from the custom name
-// section. Otherwise, if it was decoded from Text Format, this defaults to the module ID stripped of leading '$'.
-//
-// For example, if the Module was decoded from the text format `(module $math)`, the default name is "math".
-//
-// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#name-section%E2%91%A0
-// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#custom-section%E2%91%A0
-// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#modules%E2%91%A0%E2%91%A2
-func (m *Module) WithName(name string) *Module {
-	m.name = name
-	return m
+// Note: In WebAssembly language, this is a decoded, validated, and possibly also compiled module. wazero avoids using
+// the name "Module" for both before and after instantiation as the name conflation has caused confusion.
+// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#semantic-phases%E2%91%A0
+type Binary struct {
```

Comment:
maybe I would prefer `CompiledCode` to line with `FromCode` ?

Reply:
fine by me